src/lib.rs

```diff
@@ -207,395 +432,112 @@ impl Mmap {
-    /// modification timestamp) may not be updated.
-    pub fn flush(&self) -> Result<()> {
-        self.inner_mut().flush_range(self.offset, self.len)
-    }
-
-    /// Asynchronously flushes outstanding memory map view modifications to disk.
-    ///
-    /// This method initiates flushing modified pages to durable storage, but it will not wait
-    /// for the operation to complete before returning. The file's metadata (including last
-    /// modification timestamp) may not be updated.
-    pub fn flush_async(&self) -> Result<()> {
-        self.inner_mut().flush_async_range(self.offset, self.len)
-    }
-
-    /// Returns the length of the memory map view.
-    pub fn len(&self) -> usize {
-        self.len
-    }
-
-    /// Returns a shared pointer to the mapped memory.
-    ///
-    /// See `Mmap::as_slice` for invariants that must hold when dereferencing the pointer.
-    pub fn ptr(&self) -> *const u8 {
-        unsafe { self.inner().ptr().offset(self.offset as isize) }
-    }
-
-    /// Returns a mutable pointer to the mapped memory.
-    ///
-    /// See `Mmap::as_mut_slice` for invariants that must hold when dereferencing the pointer.
-    pub fn mut_ptr(&mut self) -> *mut u8 {
-        unsafe { self.inner_mut().mut_ptr().offset(self.offset as isize) }
-    }
-
-    /// Returns the memory mapped file as an immutable slice.
-    ///
-    /// ## Unsafety
-    ///
-    /// The caller must ensure that the file is not concurrently modified.
-    pub unsafe fn as_slice(&self) -> &[u8] {
-        &self.inner().as_slice()[self.offset..self.offset + self.len]
-    }
-
-    /// Returns the memory mapped file as a mutable slice.
-    ///
-    /// ## Unsafety
-    ///
-    /// The caller must ensure that the file is not concurrently accessed.
-    pub unsafe fn as_mut_slice(&mut self) -> &mut [u8] {
-        &mut self.inner_mut().as_mut_slice()[self.offset..self.offset + self.len]
-    }
-
-    /// Clones the view of the memory map.
-    ///
-    /// The underlying memory map is shared, and thus the caller must ensure that the memory
-    /// underlying the view is not illegally aliased.
-    pub unsafe fn clone(&self) -> MmapViewSync {
-        MmapViewSync {
-            inner: self.inner.clone(),
-            offset: self.offset,
-            len: self.len,
+            slice::from_raw_parts_mut(self.inner.mut_ptr(), self.inner.len())
         }
     }
 }
 
-impl fmt::Debug for MmapViewSync {
+impl fmt::Debug for MmapMut {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "MmapViewSync {{ ptr: {:?}, offset: {}, len: {} }}",
-               self.inner().ptr(), self.offset, self.len)
+        write!(f, "MmapMut {{ ptr: {:?}, len: {} }}", self.as_ptr(), self.len())
     }
 }
 
-unsafe impl Sync for MmapViewSync {}
-unsafe impl Send for MmapViewSync {}
-
 #[cfg(test)]
 mod test {
+    // memmap::file() is safe on Windows
+    #![allow(unused_unsafe)]
+
+    mod memmap {
+        pub use super::super::*;
```

Comment:
It is, I just wanted the tests code to say `memmap::file` instead of just `file`, etc. As I said above, `use super as memmap` doesn't work, this is either a bug or me misunderstanding how the module system works.